sdk/src/main/java/com/sygic/travel/sdk/auth/service/AuthService.kt

```diff
@@ -0,0 +1,154 @@
+package com.sygic.travel.sdk.auth.service
+
+import com.sygic.travel.sdk.auth.api.SygicTravelAuthApiClient
+import com.sygic.travel.sdk.auth.model.AuthorizationRequest
+import com.sygic.travel.sdk.auth.model.UserRegistrationRequest
+import com.sygic.travel.sdk.auth.model.UserRegistrationResponse
+import com.sygic.travel.sdk.auth.model.UserSession
+import kotlinx.coroutines.experimental.CommonPool
+import kotlinx.coroutines.experimental.launch
+import java.util.Date
+
+
+class AuthService(
+	private val sygicTravelAuthClient: SygicTravelAuthApiClient,
+	private val authStorageService: AuthStorageService,
+	private val clientId: String
+) {
+
+	fun authorize(authRequest: AuthorizationRequest): UserSession? {
+		val response = sygicTravelAuthClient.authorize(authRequest).execute().body() ?: return null
+
+		setAuthorizationStorage(
+			response.accessToken,
+			response.expiresIn,
+			response.refreshToken
+		)
+		return response
+	}
+
+	fun register(userRegistrationRequest: UserRegistrationRequest): UserRegistrationResponse? {
+		var clientSession = authStorageService.getClientSession()
+		if (clientSession.isNullOrBlank()) {
```

Comment:
the nullOrBlank normalization should be done in the getter

Reply:
I suppose you mean in `getClientSession()`. IMO ... don't like it very much, I would have to wire `AuthService` dependency to `AuthStorageService` which only cares about saving/retrieving data.